core/container.go

```diff
@@ -339,6 +339,24 @@ func (container *Container) WithFS(ctx context.Context, dir *Directory) (*Contai
 	return &Container{ID: id}, nil
 }
 
+func (container *Container) WithDirectory(ctx context.Context, gw bkgw.Client, subdir string, src *Directory, filter CopyFilter) (*Container, error) {
+	return container.updateRootFS(ctx, gw, subdir, func(dir *Directory) (*Directory, error) {
+		return dir.WithDirectory(ctx, "", src, filter)
+	})
+}
+
+func (container *Container) WithFile(ctx context.Context, gw bkgw.Client, subdir string, src *File) (*Container, error) {
+	return container.updateRootFS(ctx, gw, subdir, func(dir *Directory) (*Directory, error) {
+		return dir.WithFile(ctx, "", src)
+	})
+}
+
+func (container *Container) WithNewFile(ctx context.Context, gw bkgw.Client, dest string, content []byte) (*Container, error) {
+	return container.updateRootFS(ctx, gw, dest, func(dir *Directory) (*Directory, error) {
+		return dir.WithNewFile(ctx, gw, "", content)
+	})
```

Comment:
This one spooks me out the most; isn't this value supposed to be a filename? Not sure how this is working at the moment. :thinking: 

This seems to work:

```suggestion
	dir, name := filepath.Split(dest)
	return container.updateRootFS(ctx, gw, dir, func(dir *Directory) (*Directory, error) {
		return dir.WithNewFile(ctx, gw, name, content)
	})
```